### ISSUE TITLE
[3.7.0-RC1] Repeated enforced Z calibration during 7x7 bed leveling

### ISSUE THREAD

**Issue by sgoll** [prusa3d/Prusa-Firmware#1651]
Since 3.7.0-RC1 I am using the new 7x7 mesh bed leveling (with 5 measurements per position). Now I have the problem that intermittently the bed leveling done before each print detects some problem, runs the enforced Z calibration and then re-runs the bed leveling.

Sometimes, this Z calibration is enforced a second or third time before finally the bed leveling succeeds.

This also happens when nothing at all was changed: the printer was switched on all the time (so no skip in the Z steppers occurred) and the steel sheet was not replaced (so it should still be aligned just like in the first run of the mesh bed leveling).

When is the Z calibration enforced? Is there some setting to tune down the sensitivity?

**Comment by L0rdS474n** [comment]
Have this too

**Comment by JohnnyPhantom27** [comment]
Me too!

**Comment by JohnnyPhantom27** [comment]
Working on finding a cause. My buddy also has the MK3 printer with the new 3.7.0-RC1, his works fine. I decided to try the calibration wizard. It told me my motor belt was loose. I made it tighter and ran the test again, everything went great until the bed leveling. I did the 4 calibration points with paper, then put on the steel sheet as instructed, it got to the 2nd point in the test (front center of the bed) and failed. Same behavior as when I just tried printing. Got a failed: check nozzle and bed for debris. I cleaned the steel sheet, the bed, and the nozzle again. I also gave the screws holding the bed in place a snug turn with the hex key. re-trying...

**Comment by JohnnyPhantom27** [comment]
Cleaned the bed, nozzle, leveling sensor, and gave the bed screws a snugging. I get past the wizard now, but during the first layer calibration, it fails during the bed leveling, front center each time. I hope my trial and errors helps.

**Comment by PavelSindler** [comment]
Thank you all for your reports.

**Comment by JohnnyPhantom27** [comment]
Figured maybe a factory reset might help, since I did have some manual leveling adjustments in the printer before going to 3.7.0-RC1, it did not :(

**Comment by L0rdS474n** [comment]
> Figured maybe a factory reset might help, since I did have some manual leveling adjustments in the printer before going to 3.7.0-RC1, it did not :(

I also did a factory restore with no solution on my MK3S

**Comment by geeksville** [comment]
btw - From looking at the code, I don't think ya'll should bother with factory resets etc... to try and find a workaround.  It just looks like it gets confused part way through moving through the 7x7 mesh and thinks it is done (when it isn't - on mine it often happens after just the fifthish point).  I didn't look more carefully.

**Comment by Shinomorisama** [comment]
I have this problem too on my MK3.   It does the leveling fine for the test,but when starting the print it actually seems faster than the test. It took 4 tries to get through the 7x7.

**Comment by PavelSindler** [comment]
Did it always happen when 7x7 mesh was turned on? Or did somebody encounter this issue also with 3x3 mesh?

**Comment by PavelSindler** [comment]
Also can you please send me serial log from this mesh bed leveling fail? (You can use Octoprint or Pronterface or using PuTTY for example (115200 Baud, CR & NL).

**Comment by Kolous** [comment]
Same issue with MK 2.5. First 5 prints the 7x7 leveling was always sucesfull now it takes more than 10 attempts to start printing. Always failing on different point....espacially frustrating is faling on last point.

**Comment by geeksville** [comment]
@PavelSindler I'm happy to send you a log if you still need it.  It always happens in 7x7 not 3x3, and it can occur on any point (sometimes 5th, sometimes not at all, sometimes last point).

**Comment by Ghenghis** [comment]
only on 7x7 at different points tried 7 times last night had to switch back to 3x3 to print something. 


Also not sure why it dont probe far right side at all. it does probe far left no problem but alot of the right side isn't being probed - almost a whole square-right-side-whole-row on the flex sheet on far right side of bed, maybe this is normal but thats alot of area not probed on the whole bed.

**Comment by Shinomorisama** [comment]
Not sure how to get the serial log.  In the terminal in octoprint it says "Bed leveling failed. Sensor didn't trigger. Debris on nozzle? Waiting for reset".  I've had one failure now on 3x3 on point 8. On 7x7 it fails almost every time but it is very random where. Usually in the first 2 rows. Watching the probe. The light seems to flicker 3 times on the points where it is failing just like every other point.  The calibration never has any trouble.  Tried calibration cold then G80 in terminal with cold bed. Still some failures.  Tried calibration hot and running a print still failures.  G28 it seems to work but it is moving slower.

**Comment by halfluck** [comment]
Phew! glad its not just me thought i had a dodgy pinda, glad its only a FW issue :)
mine does it on 3x3 and 7x7

**Comment by Kolous** [comment]
The issues I described are with 7x7 and 5 checks. When decreasing to 4 it works reliably.

Edit: More reliable. Just had first issue with that too,

**Comment by Nippylicious** [comment]
Glad to hear that you guys had the same issues.
Mine 2.5s only works with 3x3.
7x7 occurs in the First (Front) row in an error.

**Comment by kripobs** [comment]
a simple G28 works for me if everything is cold. also the xyz calibration was ok. but as soon as i start print and everything is heated up it stops in the middle of the front and want to force calibrate z. in both modes 3x3 and 7x7
reverting back to an older firmware so no logs from me. sorry (i also dont know how to make one :D)

**Comment by sgoll** [comment]
> Also can you please send me serial log from this mesh bed leveling fail?

@PavelSindler Here is the serial output when the bed leveling fails:

```
tmc2130_goto_step 1 1 2 1000 
tmc2130_home_exit tmc2130_sg_homing_axes_mask=0x02
echo:busy: processing
(…)
echo:busy: processing
Bed leveling failed. Sensor didnt trigger. Debris on nozzle? Waiting for reset.echo:busy: processing
echo:busy: processing
echo:busy: paused for user
(…)
echo:busy: paused for user
tmc2130_home_enter(axes_mask=0x04)
```

**Comment by sgoll** [comment]
This may have to do something with the PINDA probe temperature. For me, the problem is much more severe when trying to print PETG (230/75) compared to PLA (215/60).

**Comment by cosmith71** [comment]
3x3 works _most_ of the time on my MK3.  7x7 almost always fails at least once. On my MK2.5S 7x7 is unusable and 3x3 fails so often I had to roll back to 3.6 just to get a usable machine.

**Comment by PavelSindler** [comment]
Thanks for help.

**Comment by Nippylicious** [comment]
do you guys have this behaivour?

1. first the bed touches gently the endstop
2. second the extruder toches gently the endstop
-- so point 1&2 are normal --
3. the extruder moves a bit to the right side X Axis
4. and then it "chrashes" into the X endstop before it start the mesh

i took a video and packed it into a zip
[Extruder_Endstop_Crash.zip](https://github.com/prusa3d/Prusa-Firmware/files/3002219/Extruder_Endstop_Crash.zip)

**Comment by leptun** [comment]
Yes, I also have this problem. I think this is caused by the old  X Y home offset calculated during XYZ calibration. Maybe running it again with the latest firmware fixes the issue. I didn’t have time to test this myself, but will do so when I go to the printer next.

**Comment by sgoll** [comment]
@Nippylicious @leptun I do not have this issue: the print head moves to the left-most 7x7 bed mesh point but does not crash into the X axis holder (although it comes rather close, < 1 mm I guess).

What are your XYZ cal. details at `Support` → `XYZ cal. details`? I have the following:

```
Y distance from min:
Left: 11.41mm
Right: 11.51mm

Measured skew: 0.00°

[0;0] point offset
X 0.27 mm
Y 0.91 mm
```

For the record, I was able to run the 7x7 mesh calibration fine for the last two PETG prints I did. Here's what was different:

- I flipped the flexible PEI steel sheet to its back.
- I started the prints on a cold printer (so only preheating in the start G-code).

I don't know if this may be relevant.

**Comment by PavelSindler** [comment]
Did you guys run Wizard or XYZ or Z-calibration after upgrading to 3.7.0-RC1? If you didn't, please run Z-calibration from menu "Calibration->Calibrate Z". 
Crashes are caused by mesh which was moved too much to the left side. There is another issue on that. I would like to solve just Z-levleing here in this thread. Thanks.

**Comment by sgoll** [comment]
@PavelSindler I had already run the Z calibration (from the `Calibration` menu) repeatedly. It did not have any effect on the problem with the mesh bed leveling.

**Comment by Nippylicious** [comment]
@sgoll:
my XYZ Calibration:

```
Y distance from min:
Left: 8.31mm
Right: 7.45mm

Measured skew: 0.19°
Slight skew: 0.12°
Serve skew: 0.25°

[0;0] point offset
X -5.02mm
Y -0.14mm
```

------------

@PavelSindler:
No, i did not run the Wizard after upgradting from 3.6.0 -> 3.7.0 RC1
I thought i was not nessecary (my bad).

When my actual print is done, i flash it back to RC1 and run the Wizard.
Then i give you feedback.

**Comment by Nippylicious** [comment]
@PavelSindler:
so.. print is finished and i flashed back to RC1
Then i ran the Wizard.

The "crash in X endstop" is still the same.
Then it run a whole 7x7 calibration before the Live Z-adjustment (with no problems).

When it finished the Z-adjustment, i ran (manually) a mesh bed leveling from the calibration menu.
Then it crashes again against the X endstop, and aborts after point 4 with an error (Please rotate the Z Axis against frame....), i did it... then the same procedure again (crash, point 4, error........)

**Comment by leptun** [comment]
In my case it might be because of the fat cable bundle hitting the frame

**Comment by Nippylicious** [comment]
> 
> Crashes are caused by mesh which was moved too much to the left side. There is another issue on that. I would like to solve just Z-levleing here in this thread. Thanks.

I created a new Issue
[https://github.com/prusa3d/Prusa-Firmware/issues/1674](url)

**Comment by cosmith71** [comment]
> Did you guys run Wizard or XYZ or Z-calibration after upgrading to 3.7.0-RC1? If you didn't, please run Z-calibration from menu "Calibration->Calibrate Z".

@PavelSindler 

I did the Z-calibration.  Now my MK2.5S is completing the 7x7 reliably.  No change on the MK3.  It won't complete a 7x7 without the "Some problem encountered, Z-leveling enforced..." error.  I did an XYZ and a Z calibration on the MK3.

**Comment by PavelSindler** [comment]
Here is fw (just MK3 and MK3S versions) with more information written to serial log, which should help us to debug this issue:
[fw-test.zip](https://github.com/prusa3d/Prusa-Firmware/files/3008061/fw-test.zip)
@sgoll would it be possible for you to flash it to your printer and send me information from log again?

**Comment by sgoll** [comment]
@PavelSindler Sure. Let me finish my current print job (12 hours to go) and I'll be able to send you the log output probably tomorrow.

**Comment by geeksville** [comment]
@PavelSindler if you can point me at a branch/commit with that extra debug info I
don't mind collecting the log on my printer (alas I need source because I
run a branch with a small change to work with my JLTX indirect filament
sensor).

(Sent from a phone - please ignore typos)

**Comment by PavelSindler** [comment]
It is "7x7_polishing" branch ;-) in my repository.

**Comment by wavexx** [comment]
Did the "skew" limits change from 3.6.2 to 3.7.0-rc1? I wanted to test 7x7 leveling, but curiously when running the XYZ calibration the procedure now detects a slight skew, whereas 3.6.2 (and all previous) were just saying that the axes are perpendicular.

**Comment by wavexx** [comment]
Just out of curiosity, just by rerunning the XYZ cal I went from 0.12 of skew to 0.06 (mmh).

**Comment by PavelSindler** [comment]
No, it didn't change. You may be somewhere in between slight skew and perpendicular.

**Comment by 3d-gussner** [comment]
@PavelSindler Just a thought, would it be easier if we had debug (read only) options by default?
I was thinking of a kind of gcode commands which write to serial or SD card (often already implemented):
- in the past on the MK2 i used quite often M45 V and G80 V verbosity levels to troubleshoot things.
- also having the (reading not writing) dcodes active by default would make a bug report much easier. Remember TheZeroBeast found temp issues in the PINDAv2 like that and he could post an issue report with really good data.
- 'normal' users don't know how to active the debug mode and compile firmware. They would be then able to provide needed data much easier.

As every machine can differ from what you have by hand, users with specific issues could dump these information in the issue report.

Gina from Octoprint uses it all the time to analyse issues. I know that the Einsy resources are hitting the limits quite often and cannot be compared to boards like the Rpi with 'unlimited' resources. But I already wrote a proposal how-to get 8-10kb free in another topic.

**Comment by PavelSindler** [comment]
Yes it is good point. We understand it and logging is on our todo lists already. Verbosity levels in calibrations are currently disabled to save flash unfortunately.

**Comment by geeksville** [comment]
Hi @PavelSindler here you go.

Test conditions:
- confirmed plate was clear of debris
- default 7x7 and three readings per point
- your branch
- the Prusa mesh bed leveling plugin in octoprint shows my bed as super level (.15mm max deviation)

Here's the relevant log output of a failed session:
```
Send: N9 M104 S160*107
Recv: ok
Send: N10 M190 S80*97
[...]
Recv: ok
Send: N11 G28 W*84
Recv: tmc2130_home_enter(axes_mask=0x01)
Recv:   0 step=14 mscnt= 226
Recv: tmc2130_goto_step 0 13 2 1000
Recv: tmc2130_home_exit tmc2130_sg_homing_axes_mask=0x01
Recv: tmc2130_home_enter(axes_mask=0x02)
Recv: echo:busy: processing
Recv: echo:busy: processing
Recv:   0 step=50 mscnt= 800
Recv: tmc2130_goto_step 1 49 2 1000
Recv: tmc2130_home_exit tmc2130_sg_homing_axes_mask=0x02
Recv: echo:busy: processing
Recv: echo:busy: processing
Recv: echo:busy: processing
Recv: ok
Send: N12 G80*34
Recv: [1.000000;1.000000]
Recv: after clamping: [1.455118;0.999955]
Recv: Min. Z: -10.000000
Recv: init Z = 0.360000, min_z = -10.000000
Recv: PINDA triggered at 0.162500
Recv: init Z = 0.362500, min_z = -10.000000
Recv: PINDA triggered at 0.150000
Recv: init Z = 0.350000, min_z = -10.000000
Recv: PINDA triggered at 0.160000
Recv: [35.000000;1.000000]
Recv: after clamping: [35.000000;1.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.355000, min_z = -10.000000
Recv: PINDA triggered at 0.142500
Recv: init Z = 0.342500, min_z = -10.000000
Recv: PINDA triggered at 0.145000
Recv: init Z = 0.345000, min_z = -10.000000
Recv: PINDA triggered at 0.135000
Recv: [69.000000;1.000000]
Recv: after clamping: [69.000000;1.000000]
Recv: echo:busy: processing
Recv: Min. Z: -10.000000
Recv: init Z = 0.320000, min_z = -10.000000
Recv: PINDA triggered at 0.100000
Recv: init Z = 0.300000, min_z = -10.000000
Recv: PINDA triggered at 0.090000
Recv: init Z = 0.290000, min_z = -10.000000
Recv: PINDA triggered at 0.125000
Recv: [3;0]: Z_offset = -72
Recv: Bed leveling, point: 3, calibration Z stored in eeprom: -72, calibration z: -0.562500
Recv: [103.000000;1.000000]
Recv: after clamping: [103.000000;1.000000]
Recv: Min. Z: -1.562500
Recv: init Z = 0.327500, min_z = -1.562500
Recv: PINDA triggered at 0.092500
Recv: init Z = 0.292500, min_z = -1.562500
Recv: PINDA triggered at 0.130000
Recv: init Z = 0.330000, min_z = -1.562500
Recv: PINDA triggered at 0.097500
Recv: [137.000000;1.000000]
Recv: after clamping: [137.000000;1.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.335000, min_z = -10.000000
Recv: PINDA triggered at 0.100000
Recv: init Z = 0.300000, min_z = -10.000000
Recv: PINDA triggered at 0.145000
Recv: init Z = 0.345000, min_z = -10.000000
Recv: PINDA triggered at 0.112500
Recv: [171.000000;1.000000]
Recv: after clamping: [171.000000;1.000000]
Recv: echo:busy: processing
Recv: Min. Z: -10.000000
Recv: init Z = 0.337500, min_z = -10.000000
Recv: PINDA triggered at 0.110000
Recv: init Z = 0.310000, min_z = -10.000000
Recv: PINDA triggered at 0.155000
Recv: init Z = 0.355000, min_z = -10.000000
Recv: PINDA triggered at 0.125000
Recv: [6;0]: Z_offset = -101
Recv: Bed leveling, point: 6, calibration Z stored in eeprom: -101, calibration z: -0.852500
Recv: [205.000000;1.000000]
Recv: after clamping: [205.000000;1.000000]
Recv: Min. Z: -1.852500
Recv: init Z = 0.275000, min_z = -1.852500
Recv: PINDA triggered at 0.067500
Recv: init Z = 0.267500, min_z = -1.852500
Recv: PINDA triggered at 0.060000
Recv: init Z = 0.260000, min_z = -1.852500
Recv: PINDA triggered at 0.050000
Recv: [205.000000;35.000000]
Recv: after clamping: [205.000000;35.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.275000, min_z = -10.000000
Recv: PINDA triggered at 0.057500
Recv: init Z = 0.257500, min_z = -10.000000
Recv: PINDA triggered at 0.047500
Recv: init Z = 0.247500, min_z = -10.000000
Recv: PINDA triggered at 0.040000
Recv: [171.000000;35.000000]
Recv: after clamping: [171.000000;35.000000]
Recv: echo:busy: processing
Recv: Min. Z: -10.000000
Recv: init Z = 0.342500, min_z = -10.000000
Recv: PINDA triggered at 0.110000
Recv: init Z = 0.310000, min_z = -10.000000
Recv: PINDA triggered at 0.160000
Recv: init Z = 0.360000, min_z = -10.000000
Recv: PINDA triggered at 0.135000
Recv: [137.000000;35.000000]
Recv: after clamping: [137.000000;35.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.335000, min_z = -10.000000
Recv: PINDA triggered at 0.162500
Recv: init Z = 0.362500, min_z = -10.000000
Recv: PINDA triggered at 0.132500
Recv: init Z = 0.332500, min_z = -10.000000
Recv: PINDA triggered at 0.162500
Recv: [103.000000;35.000000]
Recv: after clamping: [103.000000;35.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.330000, min_z = -10.000000
Recv: PINDA triggered at 0.095000
Recv: init Z = 0.295000, min_z = -10.000000
Recv: PINDA triggered at 0.135000
Recv: init Z = 0.335000, min_z = -10.000000
Recv: PINDA triggered at 0.100000
Recv: [69.000000;35.000000]
Recv: after clamping: [69.000000;35.000000]
Recv: echo:busy: processing
Recv: Min. Z: -10.000000
Recv: init Z = 0.317500, min_z = -10.000000
Recv: PINDA triggered at 0.095000
Recv: init Z = 0.295000, min_z = -10.000000
Recv: PINDA triggered at 0.097500
Recv: init Z = 0.297500, min_z = -10.000000
Recv: PINDA triggered at 0.087500
Recv: [35.000000;35.000000]
Recv: after clamping: [35.000000;35.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.385000, min_z = -10.000000
Recv: PINDA triggered at 0.172500
Recv: init Z = 0.372500, min_z = -10.000000
Recv: PINDA triggered at 0.170000
Recv: init Z = 0.370000, min_z = -10.000000
Recv: PINDA triggered at 0.180000
Recv: [1.000000;35.000000]
Recv: after clamping: [1.458447;34.999958]
Recv: Min. Z: -10.000000
Recv: init Z = 0.387500, min_z = -10.000000
Recv: PINDA triggered at 0.165000
Recv: init Z = 0.365000, min_z = -10.000000
Recv: big deviation
Recv: ���.\x03echo:busy: processing
Recv: echo:busy: processing
Recv: echo:busy: paused for user
Recv: echo:busy: paused for user
Recv: echo:busy: paused for user
Recv: echo:busy: paused for user
Recv: echo:busy: paused for user
Recv: echo:busy: paused for user
Recv: echo:busy: paused for user
Recv: echo:busy: paused for user
Recv: echo:busy: paused for user
Recv: echo:busy: paused for user
```

Here's the output from a successful run:
```

Recv: ok
Send: N9 M104 S160*107
Recv: ok
Send: N10 M190 S80*97
[...]
Recv: ok
Send: N11 G28 W*84
Recv: tmc2130_home_enter(axes_mask=0x01)
Recv:   0 step=14 mscnt= 224
Recv: tmc2130_goto_step 0 13 2 1000
Recv: tmc2130_home_exit tmc2130_sg_homing_axes_mask=0x01
Recv: tmc2130_home_enter(axes_mask=0x02)
Recv: echo:busy: processing
Recv: echo:busy: processing
Recv:   0 step=18 mscnt= 288
Recv: tmc2130_goto_step 1 49 2 1000
Recv: tmc2130_home_exit tmc2130_sg_homing_axes_mask=0x02
Recv: echo:busy: processing
Recv: echo:busy: processing
Recv: echo:busy: processing
Recv: ok
Send: N12 G80*34
Recv: [1.000000;1.000000]
Recv: after clamping: [1.455118;0.999955]
Recv: Min. Z: -10.000000
Recv: init Z = 0.365000, min_z = -10.000000
Recv: PINDA triggered at 0.162500
Recv: init Z = 0.362500, min_z = -10.000000
Recv: PINDA triggered at 0.165000
Recv: init Z = 0.365000, min_z = -10.000000
Recv: PINDA triggered at 0.162500
Recv: [35.000000;1.000000]
Recv: after clamping: [35.000000;1.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.342500, min_z = -10.000000
Recv: PINDA triggered at 0.137500
Recv: init Z = 0.337500, min_z = -10.000000
Recv: PINDA triggered at 0.137500
Recv: init Z = 0.337500, min_z = -10.000000
Recv: PINDA triggered at 0.135000
Recv: [69.000000;1.000000]
Recv: after clamping: [69.000000;1.000000]
Recv: echo:busy: processing
Recv: Min. Z: -10.000000
Recv: init Z = 0.292500, min_z = -10.000000
Recv: PINDA triggered at 0.080000
Recv: init Z = 0.280000, min_z = -10.000000
Recv: PINDA triggered at 0.072500
Recv: init Z = 0.272500, min_z = -10.000000
Recv: PINDA triggered at 0.065000
Recv: [3;0]: Z_offset = -72
Recv: Bed leveling, point: 3, calibration Z stored in eeprom: -72, calibration z: -0.556667
Recv: [103.000000;1.000000]
Recv: after clamping: [103.000000;1.000000]
Recv: Min. Z: -1.556667
Recv: init Z = 0.280000, min_z = -1.556667
Recv: PINDA triggered at 0.050000
Recv: init Z = 0.250000, min_z = -1.556667
Recv: PINDA triggered at 0.092500
Recv: init Z = 0.292500, min_z = -1.556667
Recv: PINDA triggered at 0.062500
Recv: [137.000000;1.000000]
Recv: after clamping: [137.000000;1.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.275000, min_z = -10.000000
Recv: PINDA triggered at 0.045000
Recv: init Z = 0.245000, min_z = -10.000000
Recv: PINDA triggered at 0.085000
Recv: init Z = 0.285000, min_z = -10.000000
Recv: PINDA triggered at 0.052500
Recv: echo:busy: processing
Recv: [171.000000;1.000000]
Recv: after clamping: [171.000000;1.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.262500, min_z = -10.000000
Recv: PINDA triggered at 0.037500
Recv: init Z = 0.237500, min_z = -10.000000
Recv: PINDA triggered at 0.080000
Recv: init Z = 0.280000, min_z = -10.000000
Recv: PINDA triggered at 0.047500
Recv: [6;0]: Z_offset = -101
Recv: Bed leveling, point: 6, calibration Z stored in eeprom: -101, calibration z: -0.846667
Recv: [205.000000;1.000000]
Recv: after clamping: [205.000000;1.000000]
Recv: Min. Z: -1.846667
Recv: init Z = 0.185000, min_z = -1.846667
Recv: PINDA triggered at -0.022500
Recv: init Z = 0.177500, min_z = -1.846667
Recv: PINDA triggered at -0.035000
Recv: init Z = 0.165000, min_z = -1.846667
Recv: PINDA triggered at -0.040000
Recv: [205.000000;35.000000]
Recv: after clamping: [205.000000;35.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.182500, min_z = -10.000000
Recv: PINDA triggered at -0.037500
Recv: init Z = 0.162500, min_z = -10.000000
Recv: PINDA triggered at -0.047500
Recv: init Z = 0.152500, min_z = -10.000000
Recv: echo:busy: processing
Recv: PINDA triggered at -0.055000
Recv: [171.000000;35.000000]
Recv: after clamping: [171.000000;35.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.257500, min_z = -10.000000
Recv: PINDA triggered at 0.090000
Recv: init Z = 0.290000, min_z = -10.000000
Recv: PINDA triggered at 0.050000
Recv: init Z = 0.250000, min_z = -10.000000
Recv: PINDA triggered at 0.030000
Recv: [137.000000;35.000000]
Recv: after clamping: [137.000000;35.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.270000, min_z = -10.000000
Recv: PINDA triggered at 0.100000
Recv: init Z = 0.300000, min_z = -10.000000
Recv: PINDA triggered at 0.067500
Recv: init Z = 0.267500, min_z = -10.000000
Recv: PINDA triggered at 0.092500
Recv: [103.000000;35.000000]
Recv: after clamping: [103.000000;35.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.280000, min_z = -10.000000
Recv: PINDA triggered at 0.052500
Recv: init Z = 0.252500, min_z = -10.000000
Recv: PINDA triggered at 0.095000
Recv: init Z = 0.295000, min_z = -10.000000
Recv: echo:busy: processing
Recv: PINDA triggered at 0.062500
Recv: [69.000000;35.000000]
Recv: after clamping: [69.000000;35.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.285000, min_z = -10.000000
Recv: PINDA triggered at 0.067500
Recv: init Z = 0.267500, min_z = -10.000000
Recv: PINDA triggered at 0.062500
Recv: init Z = 0.262500, min_z = -10.000000
Recv: PINDA triggered at 0.057500
Recv: [35.000000;35.000000]
Recv: after clamping: [35.000000;35.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.362500, min_z = -10.000000
Recv: PINDA triggered at 0.157500
Recv: init Z = 0.357500, min_z = -10.000000
Recv: PINDA triggered at 0.157500
Recv: init Z = 0.357500, min_z = -10.000000
Recv: PINDA triggered at 0.155000
Recv: [1.000000;35.000000]
Recv: after clamping: [1.458447;34.999958]
Recv: Min. Z: -10.000000
Recv: init Z = 0.372500, min_z = -10.000000
Recv: echo:busy: processing
Recv: PINDA triggered at 0.170000
Recv: init Z = 0.370000, min_z = -10.000000
Recv: PINDA triggered at 0.167500
Recv: init Z = 0.367500, min_z = -10.000000
Recv: PINDA triggered at 0.165000
Recv: [1.000000;69.000000]
Recv: after clamping: [1.461776;68.999954]
Recv: Min. Z: -10.000000
Recv: init Z = 0.372500, min_z = -10.000000
Recv: PINDA triggered at 0.172500
Recv: init Z = 0.372500, min_z = -10.000000
Recv: PINDA triggered at 0.170000
Recv: init Z = 0.370000, min_z = -10.000000
Recv: PINDA triggered at 0.167500
Recv: [35.000000;69.000000]
Recv: after clamping: [35.000000;69.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.360000, min_z = -10.000000
Recv: PINDA triggered at 0.155000
Recv: init Z = 0.355000, min_z = -10.000000
Recv: PINDA triggered at 0.152500
Recv: init Z = 0.352500, min_z = -10.000000
Recv: PINDA triggered at 0.152500
Recv: [69.000000;69.000000]
Recv: after clamping: [69.000000;69.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.295000, min_z = -10.000000
Recv: echo:busy: processing
Recv: PINDA triggered at 0.077500
Recv: init Z = 0.277500, min_z = -10.000000
Recv: PINDA triggered at 0.070000
Recv: init Z = 0.270000, min_z = -10.000000
Recv: PINDA triggered at 0.062500
Recv: [103.000000;69.000000]
Recv: after clamping: [103.000000;69.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.305000, min_z = -10.000000
Recv: PINDA triggered at 0.072500
Recv: init Z = 0.272500, min_z = -10.000000
Recv: PINDA triggered at 0.112500
Recv: init Z = 0.312500, min_z = -10.000000
Recv: PINDA triggered at 0.085000
Recv: [137.000000;69.000000]
Recv: after clamping: [137.000000;69.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.285000, min_z = -10.000000
Recv: PINDA triggered at 0.102500
Recv: init Z = 0.302500, min_z = -10.000000
Recv: PINDA triggered at 0.067500
Recv: init Z = 0.267500, min_z = -10.000000
Recv: PINDA triggered at 0.097500
Recv: [171.000000;69.000000]
Recv: after clamping: [171.000000;69.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.267500, min_z = -10.000000
Recv: echo:busy: processing
Recv: PINDA triggered at 0.042500
Recv: init Z = 0.242500, min_z = -10.000000
Recv: PINDA triggered at 0.085000
Recv: init Z = 0.285000, min_z = -10.000000
Recv: PINDA triggered at 0.057500
Recv: [205.000000;69.000000]
Recv: after clamping: [205.000000;69.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.185000, min_z = -10.000000
Recv: PINDA triggered at -0.027500
Recv: init Z = 0.172500, min_z = -10.000000
Recv: PINDA triggered at -0.032500
Recv: init Z = 0.167500, min_z = -10.000000
Recv: PINDA triggered at -0.045000
Recv: [6;3]: Z_offset = -103
Recv: Bed leveling, point: 21, calibration Z stored in eeprom: -103, calibration z: -0.866667
Recv: [205.000000;103.000000]
Recv: after clamping: [205.000000;103.000000]
Recv: Min. Z: -1.866667
Recv: init Z = 0.222500, min_z = -1.866667
Recv: PINDA triggered at 0.002500
Recv: init Z = 0.202500, min_z = -1.866667
Recv: PINDA triggered at -0.007500
Recv: init Z = 0.192500, min_z = -1.866667
Recv: PINDA triggered at -0.017500
Recv: [171.000000;103.000000]
Recv: after clamping: [171.000000;103.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.255000, min_z = -10.000000
Recv: echo:busy: processing
Recv: PINDA triggered at 0.025000
Recv: init Z = 0.225000, min_z = -10.000000
Recv: PINDA triggered at 0.062500
Recv: init Z = 0.262500, min_z = -10.000000
Recv: PINDA triggered at 0.037500
Recv: [137.000000;103.000000]
Recv: after clamping: [137.000000;103.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.250000, min_z = -10.000000
Recv: PINDA triggered at 0.085000
Recv: init Z = 0.285000, min_z = -10.000000
Recv: PINDA triggered at 0.052500
Recv: init Z = 0.252500, min_z = -10.000000
Recv: PINDA triggered at 0.080000
Recv: [3;3]: Z_offset = -51
Recv: Bed leveling, point: 24, calibration Z stored in eeprom: -51, calibration z: -0.346667
Recv: [103.000000;103.000000]
Recv: after clamping: [103.000000;103.000000]
Recv: Min. Z: -1.346667
Recv: init Z = 0.322500, min_z = -1.346667
Recv: PINDA triggered at 0.092500
Recv: init Z = 0.292500, min_z = -1.346667
Recv: PINDA triggered at 0.132500
Recv: init Z = 0.332500, min_z = -1.346667
Recv: PINDA triggered at 0.102500
Recv: [69.000000;103.000000]
Recv: after clamping: [69.000000;103.000000]
Recv: echo:busy: processing
Recv: Min. Z: -10.000000
Recv: init Z = 0.285000, min_z = -10.000000
Recv: PINDA triggered at 0.067500
Recv: init Z = 0.267500, min_z = -10.000000
Recv: PINDA triggered at 0.060000
Recv: init Z = 0.260000, min_z = -10.000000
Recv: PINDA triggered at 0.055000
Recv: [35.000000;103.000000]
Recv: after clamping: [35.000000;103.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.335000, min_z = -10.000000
Recv: PINDA triggered at 0.115000
Recv: init Z = 0.315000, min_z = -10.000000
Recv: PINDA triggered at 0.115000
Recv: init Z = 0.315000, min_z = -10.000000
Recv: PINDA triggered at 0.112500
Recv: [0;3]: Z_offset = 22
Recv: Bed leveling, point: 27, calibration Z stored in eeprom: 22, calibration z: 0.383333
Recv: [1.000000;103.000000]
Recv: after clamping: [1.465105;102.999950]
Recv: Min. Z: -0.616667
Recv: init Z = 0.392500, min_z = -0.616667
Recv: PINDA triggered at 0.182500
Recv: init Z = 0.382500, min_z = -0.616667
Recv: PINDA triggered at 0.180000
Recv: init Z = 0.380000, min_z = -0.616667
Recv: PINDA triggered at 0.175000
Recv: [1.000000;137.000000]
Recv: after clamping: [1.468434;136.999950]
Recv: echo:busy: processing
Recv: Min. Z: -10.000000
Recv: init Z = 0.385000, min_z = -10.000000
Recv: PINDA triggered at 0.187500
Recv: init Z = 0.387500, min_z = -10.000000
Recv: PINDA triggered at 0.192500
Recv: init Z = 0.392500, min_z = -10.000000
Recv: PINDA triggered at 0.192500
Recv: [35.000000;137.000000]
Recv: after clamping: [35.000000;137.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.382500, min_z = -10.000000
Recv: PINDA triggered at 0.180000
Recv: init Z = 0.380000, min_z = -10.000000
Recv: PINDA triggered at 0.177500
Recv: init Z = 0.377500, min_z = -10.000000
Recv: PINDA triggered at 0.177500
Recv: [69.000000;137.000000]
Recv: after clamping: [69.000000;137.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.320000, min_z = -10.000000
Recv: PINDA triggered at 0.102500
Recv: init Z = 0.302500, min_z = -10.000000
Recv: PINDA triggered at 0.092500
Recv: init Z = 0.292500, min_z = -10.000000
Recv: PINDA triggered at 0.087500
Recv: [103.000000;137.000000]
Recv: after clamping: [103.000000;137.000000]
Recv: echo:busy: processing
Recv: Min. Z: -10.000000
Recv: init Z = 0.320000, min_z = -10.000000
Recv: PINDA triggered at 0.087500
Recv: init Z = 0.287500, min_z = -10.000000
Recv: PINDA triggered at 0.127500
Recv: init Z = 0.327500, min_z = -10.000000
Recv: PINDA triggered at 0.100000
Recv: [137.000000;137.000000]
Recv: after clamping: [137.000000;137.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.322500, min_z = -10.000000
Recv: PINDA triggered at 0.090000
Recv: init Z = 0.290000, min_z = -10.000000
Recv: PINDA triggered at 0.132500
Recv: init Z = 0.332500, min_z = -10.000000
Recv: PINDA triggered at 0.100000
Recv: [171.000000;137.000000]
Recv: after clamping: [171.000000;137.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.295000, min_z = -10.000000
Recv: PINDA triggered at 0.070000
Recv: init Z = 0.270000, min_z = -10.000000
Recv: PINDA triggered at 0.112500
Recv: init Z = 0.312500, min_z = -10.000000
Recv: PINDA triggered at 0.087500
Recv: [205.000000;137.000000]
Recv: after clamping: [205.000000;137.000000]
Recv: echo:busy: processing
Recv: Min. Z: -10.000000
Recv: init Z = 0.230000, min_z = -10.000000
Recv: PINDA triggered at 0.015000
Recv: init Z = 0.215000, min_z = -10.000000
Recv: PINDA triggered at 0.010000
Recv: init Z = 0.210000, min_z = -10.000000
Recv: PINDA triggered at 0.000000
Recv: [205.000000;171.000000]
Recv: after clamping: [205.000000;171.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.232500, min_z = -10.000000
Recv: PINDA triggered at 0.022500
Recv: init Z = 0.222500, min_z = -10.000000
Recv: PINDA triggered at 0.022500
Recv: init Z = 0.222500, min_z = -10.000000
Recv: PINDA triggered at 0.062500
Recv: [171.000000;171.000000]
Recv: after clamping: [171.000000;171.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.267500, min_z = -10.000000
Recv: PINDA triggered at 0.092500
Recv: init Z = 0.292500, min_z = -10.000000
Recv: PINDA triggered at 0.062500
Recv: init Z = 0.262500, min_z = -10.000000
Recv: PINDA triggered at 0.095000
Recv: [137.000000;171.000000]
Recv: after clamping: [137.000000;171.000000]
Recv: echo:busy: processing
Recv: Min. Z: -10.000000
Recv: init Z = 0.295000, min_z = -10.000000
Recv: PINDA triggered at 0.127500
Recv: init Z = 0.327500, min_z = -10.000000
Recv: PINDA triggered at 0.097500
Recv: init Z = 0.297500, min_z = -10.000000
Recv: PINDA triggered at 0.122500
Recv: [103.000000;171.000000]
Recv: after clamping: [103.000000;171.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.290000, min_z = -10.000000
Recv: PINDA triggered at 0.062500
Recv: init Z = 0.262500, min_z = -10.000000
Recv: PINDA triggered at 0.102500
Recv: init Z = 0.302500, min_z = -10.000000
Recv: PINDA triggered at 0.070000
Recv: [69.000000;171.000000]
Recv: after clamping: [69.000000;171.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.320000, min_z = -10.000000
Recv: PINDA triggered at 0.097500
Recv: init Z = 0.297500, min_z = -10.000000
Recv: PINDA triggered at 0.090000
Recv: init Z = 0.290000, min_z = -10.000000
Recv: PINDA triggered at 0.127500
Recv: [35.000000;171.000000]
Recv: after clamping: [35.000000;171.000000]
Recv: echo:busy: processing
Recv: Min. Z: -10.000000
Recv: init Z = 0.390000, min_z = -10.000000
Recv: PINDA triggered at 0.180000
Recv: init Z = 0.380000, min_z = -10.000000
Recv: PINDA triggered at 0.180000
Recv: init Z = 0.380000, min_z = -10.000000
Recv: PINDA triggered at 0.177500
Recv: [1.000000;171.000000]
Recv: after clamping: [1.471763;170.999950]
Recv: Min. Z: -10.000000
Recv: init Z = 0.410000, min_z = -10.000000
Recv: PINDA triggered at 0.207500
Recv: init Z = 0.407500, min_z = -10.000000
Recv: PINDA triggered at 0.205000
Recv: init Z = 0.405000, min_z = -10.000000
Recv: PINDA triggered at 0.202500
Recv: [0;6]: Z_offset = 20
Recv: Bed leveling, point: 42, calibration Z stored in eeprom: 20, calibration z: 0.363333
Recv: [1.000000;205.000000]
Recv: after clamping: [1.475092;204.999950]
Recv: Min. Z: -0.636667
Recv: init Z = 0.405000, min_z = -0.636667
Recv: PINDA triggered at 0.200000
Recv: init Z = 0.400000, min_z = -0.636667
Recv: PINDA triggered at 0.200000
Recv: init Z = 0.400000, min_z = -0.636667
Recv: PINDA triggered at 0.197500
Recv: [35.000000;205.000000]
Recv: after clamping: [35.000000;205.000000]
Recv: echo:busy: processing
Recv: Min. Z: -10.000000
Recv: init Z = 0.340000, min_z = -10.000000
Recv: PINDA triggered at 0.130000
Recv: init Z = 0.330000, min_z = -10.000000
Recv: PINDA triggered at 0.127500
Recv: init Z = 0.327500, min_z = -10.000000
Recv: PINDA triggered at 0.125000
Recv: [69.000000;205.000000]
Recv: after clamping: [69.000000;205.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.275000, min_z = -10.000000
Recv: PINDA triggered at 0.057500
Recv: init Z = 0.257500, min_z = -10.000000
Recv: PINDA triggered at 0.047500
Recv: init Z = 0.247500, min_z = -10.000000
Recv: PINDA triggered at 0.042500
Recv: [3;6]: Z_offset = -53
Recv: Bed leveling, point: 45, calibration Z stored in eeprom: -53, calibration z: -0.366667
Recv: [103.000000;205.000000]
Recv: after clamping: [103.000000;205.000000]
Recv: Min. Z: -1.366667
Recv: init Z = 0.275000, min_z = -1.366667
Recv: PINDA triggered at 0.045000
Recv: init Z = 0.245000, min_z = -1.366667
Recv: PINDA triggered at 0.087500
Recv: init Z = 0.287500, min_z = -1.366667
Recv: PINDA triggered at 0.057500
Recv: [137.000000;205.000000]
Recv: after clamping: [137.000000;205.000000]
Recv: echo:busy: processing
Recv: Min. Z: -10.000000
Recv: init Z = 0.272500, min_z = -10.000000
Recv: PINDA triggered at 0.040000
Recv: init Z = 0.240000, min_z = -10.000000
Recv: PINDA triggered at 0.080000
Recv: init Z = 0.280000, min_z = -10.000000
Recv: PINDA triggered at 0.047500
Recv: [171.000000;205.000000]
Recv: after clamping: [171.000000;205.000000]
Recv: Min. Z: -10.000000
Recv: init Z = 0.247500, min_z = -10.000000
Recv: PINDA triggered at 0.020000
Recv: init Z = 0.220000, min_z = -10.000000
Recv: PINDA triggered at -0.002500
Recv: init Z = 0.197500, min_z = -10.000000
Recv: PINDA triggered at 0.042500
Recv: [6;6]: Z_offset = -102
Recv: Bed leveling, point: 48, calibration Z stored in eeprom: -102, calibration z: -0.856667
Recv: [205.000000;205.000000]
Recv: after clamping: [205.000000;205.000000]
Recv: Min. Z: -1.856667
Recv: init Z = 0.217500, min_z = -1.856667
Recv: PINDA triggered at 0.002500
Recv: init Z = 0.202500, min_z = -1.856667
Recv: PINDA triggered at -0.007500
Recv: init Z = 0.192500, min_z = -1.856667
Recv: PINDA triggered at -0.015000
Recv: echo:busy: processing
Recv: echo:busy: processing
Recv: ok
```

Here's the gcode I used (makes it easier to retest rather than printing a model):
```
M201 X1000 Y1000 Z1000 E9000 ; sets maximum accelerations, mm/sec^2
M203 X200 Y200 Z12 E120 ; sets maximum feedrates, mm/sec
M204 S1250 T1250 ; sets acceleration (S) and retract acceleration (T)
M205 X8 Y8 Z0.4 E1.5 ; sets the jerk limits, mm/sec
M205 S0 T0 ; sets the minimum extruding and travel feed rate, mm/sec
G90 ; use absolute positions (for octolapse) kevinh
M83  ; extruder relative mode
M104 S160 ; kevinh preheat extruder partially - don't wait - but a low temp to prevent drooling
M190 S80 ; set and wait for bed temp
G28 W ; home all without mesh bed level
G80 ; mesh bed leveling
@BEDLEVELVISUALIZER
G81 ; kevinh send mesh level results to octoprint plugin

M107 ; turn off fan
G1 Z50 X0 Y200; home X axis
M104 S0 ; turn off temperature
M140 S0 ; turn off heatbed
M84 ; disable motors
```

**Comment by cosmith71** [comment]
> > Did you guys run Wizard or XYZ or Z-calibration after upgrading to 3.7.0-RC1? If you didn't, please run Z-calibration from menu "Calibration->Calibrate Z".
> 
> @PavelSindler
> 
> I did the Z-calibration. Now my MK2.5S is completing the 7x7 reliably. No change on the MK3. It won't complete a 7x7 without the "Some problem encountered, Z-leveling enforced..." error. I did an XYZ and a Z calibration on the MK3.

Spoke too soon.  It won't complete a 7x7 or a 3x3 this morning.  Re-doing XYZ calibration to see if that helps.

**Comment by sgoll** [comment]
@PavelSindler I managed to flash the modified firmware and reproduced the error (in the last few prints the error did not happen as often as previously, so it took me some attempts but then it happened again).

I am attaching two runs: one that finished mesh bed leveling without problems and one that failed (with an error of "big deviation").

- [prusa_good.txt](https://github.com/prusa3d/Prusa-Firmware/files/3012711/prusa_good.txt)
- [prusa_bad.txt](https://github.com/prusa3d/Prusa-Firmware/files/3012712/prusa_bad.txt)

**Comment by PavelSindler** [comment]
Actually it seems now that this issue happens when printer is placed on table which resonates during mesh bed leveling Z-probing. We were able to replicate this issue on two printers. These printers were failing during mesh bed leveling with the same error message that was reported by you (high deviation between currently measured Z-coordinate and avg Z value from previous measurements). When it was moved to different table, deviation significantly decreased. Do you have your printers on stable table?

**Comment by geeksville** [comment]
@PavelSindler yes.  This table (the rolling wheels are locked and on a hardwood floor).  The feet I'm using are not the stock prusa rubber feet (because those kept coming off) - rather PETG + M3 bolt with felt pads (see photo).  
![IMG_20190327_113533](https://user-images.githubusercontent.com/225513/55103162-32859b00-5085-11e9-89f8-e3afe551fe40.jpg)

**Comment by geeksville** [comment]
Perhaps fix by a small delay after the move? 50ms?

**Comment by PavelSindler** [comment]
You can try this fw, which I believe should solve the issue:
[PavelSindler_fw_mk3-3.5.3+7x7_polishing.2179.g58632975.201903271926.zip](https://github.com/prusa3d/Prusa-Firmware/files/3014973/PavelSindler_fw_mk3-3.5.3%2B7x7_polishing.2179.g58632975.201903271926.zip)
When this issue with high deviation occurs for the first time, current point measurement is repeated with old/standard Z movements. If it fails also, Z-leveling is enforced.

**Comment by sgoll** [comment]
@PavelSindler Thanks, I'll try that firmware after my current print has finished.

Regarding the unstable table: Not really. I have my printer on a quite sturdy wooden bench-like table, using the Prusa-supplied rubber feet. There is no noticeable vibration on the table when the printer is moving its axes.

**Comment by PavelSindler** [comment]
Ok, so this theory is probably wrong.

**Comment by geeksville** [comment]
Hmm.  Could it be the pinda probe getting slightly different readings due
to a small amount of inductive current generated each time the bed heater
is toggled on/off?  Ie perhaps compare to a build that briefly disables the
bed heater before each reading?

(Sent from a phone - please ignore typos)

**Comment by sgoll** [comment]
That would explain why I never seem to get this error when I manually run the 7x7 calibration from the Calibration menu but almost always when running as part of the start G-code.

**Comment by geeksville** [comment]
And just musing a bit more on this theory: perhaps it is worse in 3.7 not
just because of the 7x7 levelling but the change to shorten the period on
the bed heater PID loop (which works great to solve the power supply
clicking).  More edges on that square wave == more chance of doing a probe
while an edge occurs?

**Comment by geeksville** [comment]
It also might explain why people on reddit are saying it happens more often
with PETG (because the bed heater gets kicked on more often for higher bed
temps)

**Comment by PavelSindler** [comment]
I don't see any obvious difference in deviation between Z-measuremets with heatbed off (target temp 0) and preheated (tried 60 C and 100C).

**Comment by sgoll** [comment]
@PavelSindler I have now run five or six prints with your firmware and have not seen any issues with the mesh bed leveling. Twice the leveling stumbled and took longer to complete but that would be your workaround, I guess.

The only thing that's somewhat annoying is that the firmware keeps reminding me at the start of every print that I should upgrade to version 3.6.0 but I think I can get rid of that in the start G-code.

**Comment by cimoalpacino** [comment]
If I'm not wrong, you should disable the M115 command in the start gcode sequence in Slic3r

**Comment by primalmotion** [comment]
to me this has been happening since 3.5.something. Getting z leveling pass is just luck. always fails on point 6 or 7

**Comment by PavelSindler** [comment]
Yes, I think it is ready for release. Our testers wanted to do more tests on mesh bed leveling today, but it seems ok and hopefully will be released today.

**Comment by 3d-gussner** [comment]
@PavelSindler don't release anything on April 1st ... People may think it is an April fool :-)

**Comment by PavelSindler** [comment]
@3d-gussner I 've just released it! If there is some nasty bug I will pretend that this release is just a  joke :-D

**Comment by fbujold** [comment]
I have the same issue, repeated Z enforcement, (MK2.5 No IR sensor, Firmware 3.7.0, issuing m80 with pronteface)
Did check the wire bundle behind extruder, isolated the probe wire, pulled stretch wiggle while checking z probe status in firmware: nothing. 
Check the frame squareness: ok
Reverted back to 3x3 with 3.7.0; same. 
symptoms, 
- at point 4 and 5 hesitation with retry:passes (hesitation is probes the point, raises higher, reprobes the goes to next)
- heat up bed at 60 :passes with previous hesitation
- heatup nozzle at 210 (PLA) starts failing at point 6 or 7 with previous hesitation
- cool back down mesh passes with hesitation at point 4 and 5

retried heating up at 210 and bed at 60,: failed.
retried **while hot**, no power on bed 59 deg or so: passed
retried cooled down: passed

Could the bed current influence the probe?
https://photos.app.goo.gl/hcJtY89ey5JvBw2b6
https://photos.app.goo.gl/RCgdMkS5ZzxxvSdp7

**Comment by michalxfanta** [comment]
Does this problem remain with FW 3.7.0?

**Comment by fbujold** [comment]
Yes it does.

Le ven. 5 avr. 2019 09 h 59, Michal Fanta <notifications@github.com> a
écrit :

**Comment by jackie-t** [comment]
Please allow users to turn off this check completely - it's a real nightmare.

As a developer, you must learn to "kill your baby" even if it hurts. This "feature" is annoying, buggy, and I have to skip mesh leveling to print (which it does fine, so that alone tells me the check is bogus), so I don't encounter "some problem" (whatever that might be).

**Comment by cosmith71** [comment]
MK2.5S and MK3 MMU2S here.  I have not had a forced Z calibration occur with the 3.7.0 final firmware.  It still has to recheck several points on both machines, but it doesn't stop.

**Comment by leptun** [comment]
I also have a Mk3, a Mk3R (modded Mk2.5 that is very close to a mk3) and soon to have Mk2.5S. I did not have any problems on the mk3; no point rechecked. On the Mk3R sometimes a point is rechecked. Hope I won’t have this problem on my Mk2.5S

**Comment by Nippylicious** [comment]
I have no Problem with 3.7.0 (MK2.5s | Bear).

Like cosmith71 said, it has to recheck on two points but it doesn't stop and works well.

(On 3.7.0-RC1 it stopped on point 5)

**Comment by phooji** [comment]
@PavelSindler I think I'm seeing behavior consistent with @geeksville's theory about the heatbed:

* Prusa MK2.5S newly upgraded
* 3.70
* Bed leveling consistently fails during 1st layer calibration:
  * 7x7: 1st or 2nd point on the back row
  * 3x3: 1st or 2nd point on the back row
* **Workaround**
   * use 'tune' to set bed temperature to ~10C lower just before starting (10C is a large enough delta so that the PID doesn't kick in before bed leveling reaches the back row)
   * bed leveling consistently succeeds at those same points when the bed heater LED is dark

Given the consistency (100% failure when the bed is maintaining steady temperature; 100% success when the bed is cold or cooling down; N=~20), I'm thinking this is looking like interference between PINDA and the bed. At least that seems much more likely than a geometry/bed property.

**Comment by fbujold** [comment]
Same behavior on an MK2.5 without S upgrade and 3.7.0 not the RC1

**Comment by ergo22** [comment]
Thank you @PavelSindler and @geeksville, I'm now pretty sure this is the issue at least for me. 

I've been having issues with the Z-calibration before the print start since about FW ver 3.5.x  I've done all the FW upgrades in between, numerous full resets, recalibration etc etc. 
For me the Calibration -> Z axis  would always work flawless on cooled down machine but at job start it did not. Also it's always the back row points that fail and typically point 8 or 9. 

Moving the Z-cal to before bed heating worked, but then the calibration accuracy seemed to be off.

Now during weekend I replaced the PINDA v2 probe with new one from Prusa that I had ordered only to find the issue to come back.

With this new theory that bed heating being active is what causing this is the first one where I can repro both the issue and the fix basically 100%. If I make sure the bed is not actively heated at all during the 9 point cal it goes through without any "double tries" or fail. As soon as I have some heater activity it'll start to retry the point 7 / 8 / 9 and sometimes fail, sometimes go through.

This would also explain why some experience this more than the others as it'll depend on bed temp at starts and maybe the production patch of power supply, cabling runs etc. 
I sometimes got it to print just by moving the print head cable bundle and thus though it would be a faulty PINDA v2, but as said the new ones behaves the same. So might be that cable moving helped as thats where the interference pickup happens at least partially.

Here is an excerpt from Prusa change log... quite possible that this was the point where my issues started...
Firmware 3.5.2-FINAL
Heatbed regulation improved:
Heatbed PWM now works on a higher frequency. This helps to reduce the power supplies "clicking" noises.

Hmm... the power cabling from main supply to the Rambo board also pass under the Z cal points 7/8/9, so perhaps I should try to shield those cables to see if that makes a difference.... or add a capacitor to the heatbed connector.

**Comment by adrianluben** [comment]
same issues here :( , for me it works with 7x7 mesh only if I select 1 measurement / point.
is anybody working on a solution?

**Comment by leptun** [comment]
I also sometimes encounter mesh bed leveling problems on my modded MK2.5S, but I never had repeated Z leveling happening.

Some info: my printer is a MK2.5S with TMC2130 drivers added by me for crash detection. This might not be relevant, but just wanted to point this out. What might influence things is the fact that I use some MK3 parts on it for the Z axis (Z Motor Holders, X motor holder, X idler, Z top parts.).

Something I observed after upgrading to the MK2.5S is that the new pinda holder is much more wobbly than on the MK2, MK2.5 and MK3.

Now to the part where I am havig issues. For some reason when I power off the machine or reset it the left Z motor spins something like 45-60degrees and gets the axis out of sync. It might be because of the weight of the motor and the way TMC drivers handle disabled motors while being powered. Another possibility is that I might have assembled something skewed on the Z axis? I actually have no idea why that happens. Anyways, when this happens and it fails mesh bed leveling I am prompted just once for Z leveling and since I have TMC drivers it happens automatically.

One more thing to note is that I also noticed what @ergo22 said about mesh bed leveling with bed on causing more points to double check. I don't think it's the power cables causing the issues since they are stationary and should cause all points to have problems.

**Comment by ergo22** [comment]
I've described a trial [here ](https://forum.prusaprinters.org/forum/original-prusa-i3-mk2.5s-mk2.5-hardware-firmware-and-software-help/3.7.0-z-leveling-enforced-error/)where I moved the power supply to the other side of the printer, so the cables would not cross under back row of calibration points.... and it did improve matters. So it could be the EMI field around the power cables that causes this...

**Comment by fbujold** [comment]
My experiments with this are; turned off power to heat bed, manually start a mesh level, passes without a retry. Would anyone validate this pls.

**Comment by ergo22** [comment]
Similar process worked for me as well. At start of the print I would allow the printer to warm up both the bed and hotend. Then as it starts the calibration routine I went into the settings and reduced Bed temp by 15 degrees for example. That would stop the bed heating for a number of seconds that is usually enough to get Z-cal done. Once it does succeed I go back to menu and restore the bed temp to what is was before. Usually it does not drop more than few degrees due to this and there is no ill effect on print.

Moving the power supply over removed the need to do the above trickery.

**Comment by leptun** [comment]
This explains why on my MK3 with enclosure and external power supply I have no problem, but on the MK2.5S with power supply in standard position it retries on many points

**Comment by Panayiotis-git** [comment]
Interesting! And since the MK2.5 has almost double the current of the heated bed, the impact to the magnetic flux is (IIRC) almost double

**Comment by leptun** [comment]
Do you think high humidity might also influence things? Where I am using the MK2.5S the humidity is kinda high. The threaded rods have started to rust

**Comment by fbujold** [comment]
I doubt the humidity has anything to do with the phenomenon. If you have any way of testing in a low humidity environment it would confirm.

**Comment by plug-it-in** [comment]
I have this problem , i3 MK3 upgraded to MK3S , MMU2 upgraded to MMU2S . FW 3.7.0

**Comment by Kachidoki2807** [comment]
Hi,
I am rising up back this issue, on the version 3.8.0 my MK2.5S is unable to achieve a complete mesh bed leveling without triggering an "enforced Z" error. On my MK3S/MMU2S, there are a one point with the slow retries but it always complete the full sequence. My MK3 (non-S) seems not affected.

I have done extensive tests, with a short G-Code doing mesh bed leveling in loop. Below the main different setup I tried, each time I do at least 10 mesh bed leveling 7x7x5 in a row:
- Only leveling, machine cold, PASS
- Bed temperature = ambiant +5°C, while at full power, PASS
- Bed temperature = ambiant +5°C, while regulating, PASS with retry
- Bed temperature = 60°C, while at full power, PASS
- Bed temperature = 60°C, while regulating, FAIL
- Bed preheated at 70°C then shuted off, PASS
- Retry the FAILED test with 3x3 grid and 3 probe, FAIL
- Retry the FAILED test with the PINDA near or far of the heatbed while preheating to discriminate a PINDA temperature issue (with or without compensation), it FAILED as well.


I have oriented those test to eliminate the temperature variable and also a mechanical dilatation issue. I have tried on my desk and on the floor. I have also eliminated the temperature compensation of the PINDA from the equation. In a previous test I have also tried to change the PINDA height, and it is definitely not related. And finally I have also eliminated the grid size itself, magnet compensation as well.

So my conclusion is: **the PWM has a negative impact on the PINDA measurement**. On my both printers, if I cutoff the heatbed just before to do the leveling and then I re-power it up, the complete sequence go through without a hitch.

I can easily assume that even if the problem is not visible for most of the users, **it might have an impact on the first layer quality** as it disturbe the measurement precision.

The easy workaround is to modify the Start G-Code to shut down the heatbead while probing, 
I let the real fix at the appreciation of the Prusa's developers.

If more tests are needed, let me know.

**Comment by wavexx** [comment]
On Wed, Sep 25 2019, Kachidoki wrote:
> So my conclusion is: **the PWM has a negative impact on the PINDA
> measurement**.

I concur PWM might be a cause of PINDA interference.

On a MK3 I also had some random, repeated measurements during 7x7 in
only FW versions 3.7 to 3.8 RC1 which happened always in the same spot.
However, the final 3.8 with the improved PWM never caused any repeated
measurements so far.

I don't have experience with the MK2.5S to say if the heatbed PCB has
changed (possibly influencing the emf).

Do you have by chance more than a single steel/textured sheet?
Does swapping or flipping them on the other side change anything?

**Comment by Kachidoki2807** [comment]
Nice to meet you Yuri,
The MK2.5S has a 12V power supply, thus the heatbed draw much more current, that IMHO could make the situation worse regarding the EMF produced. I also noticed that the electronic is a bit sensitive to the EMC while just placing the steel sheet on the heatbed can sometime lead to a blank LCD.

I have just tried your suggestion and redone the basic test (60°C + MBL through the calibration menu), only once for each iteration, with 4 differents steel sheet, 3 smooth and 1 textured, on both side for each (pfew, took me half an hour). None of them has completed properly.

Maybe the title of this issue should be re-evaluated as it is still present in 3.8.0 and not only while using the 7x7 grid.

**Comment by Kachidoki2807** [comment]
Hello,
I have done more tests as explained [here](https://github.com/prusa3d/PrusaSlicer-settings/issues/80#issuecomment-536448826).
The workaround for stuck users (by modifying the Start G-Code) is only a workaround and it is not usable for a production profile. On high temp, the heatbed loose too much heat.
So I suggest the firmware might do a more surgical operation by disabling the PWM between each measurement point.

**Comment by jerm** [comment]
I started experiencing this today, after a month+ of perfect performance on a kit mk3s.

Interestingly, trying with a cold bed and no bed power is also failing reliably for me :( 

FW 3.7.2-2363

Z-Calibration does pass with flying colors, tho :-/

**Comment by leptun** [comment]
@jerm Can you please post a video of it? I’m mostly interested in seeing the nozzle and the pinda from the front of the printer. Your issue could be caused by something else.

**Comment by jerm** [comment]
I updated to 3.8.0 and..... it's better. It completes now, but it definitely has to retry in a couple of spots that it never did before. I can post video of this, tho.

Timing-wise, the only thing that correlates is that we had an earthquake this morning during a print, which gave me about 2mm of lateral layer shift for the rest of the print. I re-ran calibration, but the above problem materialized

**Comment by jerm** [comment]
@leptun Video: http://share.jermops.com/61d892f5221e . Might have to download if it doesn't work in-browser.

P.S. thanks for taking a look! :)

**Comment by leptun** [comment]
@jerm Wow. Those are a lot of retries. Something weird I noticed is that it is performing it really fast. Have you increased the acceleration? Please post the M503 output right after boot.

**Comment by leptun** [comment]
I'm trying to make a fix by quickly disabling PWM before each point. However, since I'm not experiencing any of the issues mentioned I need some volunteers for this test. Below you can see attached a zip with binaries for all printers. @Kachidoki2807 @plug-it-in @ergo22 @jerm 

Edit: New binaries. Now the bed stays either ON or OFF during probing so no bed switch occurs. I noticed that the bed would sometimes stay ON during probing with bed at 100C and would disable itself during one of the points. That's why the binaries changed. 
[FW381-RC1-Build2851.zip](https://github.com/prusa3d/Prusa-Firmware/files/3694046/FW381-RC1-Build2851.zip)

**Comment by jerm** [comment]
Haven't touched acceleration :-/ 

Send: M503
Recv: echo:Steps per unit:
Recv: echo:  M92 X100.00 Y100.00 Z400.00 E280.00
Recv: echo:UStep resolution:
Recv: echo: M350 X16 Y16 Z16 E32
Recv: echo:Maximum feedrates - normal (mm/s):
Recv: echo:  M203 X200.00 Y200.00 Z12.00 E120.00
Recv: echo:Maximum feedrates - stealth (mm/s):
Recv: echo:  M203 X100.00 Y100.00 Z12.00 E120.00
Recv: echo:Maximum acceleration - normal (mm/s2):
Recv: echo:  M201 X1000 Y1000 Z200 E5000
Recv: echo:Maximum acceleration - stealth (mm/s2):
Recv: echo:  M201 X960 Y960 Z200 E5000
Recv: echo:Acceleration: S=acceleration, T=retract acceleration
Recv: echo:  M204 S1250.00 T1250.00
Recv: echo:Advanced variables: S=Min feedrate (mm/s), T=Min travel feedrate (mm/s), B=minimum segment time (ms), X=maximum XY jerk (mm/s),  Z=maximum Z jerk (mm/s),  E=maximum E jerk (mm/s)
Recv: echo:  M205 S0.00 T0.00 B0.00 X10.00 Y10.00 Z0.40 E2.50
Recv: echo:Home offset (mm):
Recv: echo:  M206 X0.00 Y0.00 Z0.00
Recv: echo:PID settings:
Recv: echo:   M301 P16.13 I1.16 D56.23
Recv: echo:PID heatbed settings:
Recv: echo:   M304 P126.13 I4.30 D924.76
Recv: echo:Retract: S=Length (mm) F:Speed (mm/m) Z: ZLift (mm)
Recv: echo:   M207 S3.00 F2700.00 Z0.00
Recv: echo:Recover: S=Extra length (mm) F:Speed (mm/m)
Recv: echo:   M208 S0.00 F480.00
Recv: echo:Auto-Retract: S=0 to disable, 1 to interpret extrude-only moves as retracts or recoveries
Recv: echo:   M209 S0
Recv: echo:Filament settings: Disabled

**Comment by leptun** [comment]
@jerm Hmmm. That is true indeed. Do you have another pinda to test? maybe it's faulty or the cable is faulty. Also try doing a Z calibration and see how much the Z motors were out of sync (you will notice it when it gets at the top). I noticed that if the Z axis is skewed and the pinda is not perfectly perpendicular to the bed it might trigger in unexpected ways during Mesh bed leveling.

Edit: nevermind. You said it passed successfully. Still check if the pinda is perpendicular in the holder.

**Comment by Kachidoki2807** [comment]
Hi @leptun,

> I'm trying to make a fix by quickly disabling PWM before each point. However, since I'm not experiencing any of the issues mentioned I need some volunteers for this test. Below you can see attached a zip with binaries for all printers. @Kachidoki2807 @plug-it-in @ergo22 @jerm
> 
> Edit: New binaries. Now the bed stays either ON or OFF during probing so no bed switch occurs. I noticed that the bed would sometimes stay ON during probing with bed at 100C and would disable itself during one of the points. That's why the binaries changed.
> [FW381-RC1-Build2851.zip](https://github.com/prusa3d/Prusa-Firmware/files/3694046/FW381-RC1-Build2851.zip)

I've just done this test on my "faulty" MK2.5S (Rambo1.3a):
- Flashed the printer with your patch
- Performed 20 MBL in a row (twice, bed at 60°C and at 85°C), completed successfuly without a single retry
- Re-flashed the official 3.8.1rc1 to check as the ambiant temperature is really different today
- The very first MBL failed on the same point as before.

Conclusion, this patch seems to handle nicely this issue. And the heatbed temperature is stable enough, even at 85°C (I avoid to go up to 100°C on this 12V printer). It could be nice if I find the time to do some more advanced tests, like to compare the MBL output values with and without the PWM and with your patch. But I can't promise it.

Now we have to take care on the side effect. For instance, what happens if the MBL fail for another reason (mechanical) while the heatbead is ON? Is the PID proprely resumed?
Do you have a link on the commit? I am curious to do a code review on it.

**Comment by leptun** [comment]
@Kachidoki2807 Thanks for testing. Here are the changes: https://github.com/leptun/Prusa-Firmware/tree/MK3_BED_LEVELING_PWM
In theory it should disable the "disable" right after the function that probes returns. This was just a quick test to see if this implementation actually fixes the problem. The only places that might need checking is what happens if `kill` is triggered during probing. It might cause undesired behavior. Testing the safety aspects is definitely needed.

**Comment by Kachidoki2807** [comment]
I have done a quick attempt to compare the MBL output values with and without the PWM.
Warning: this test is not revelant since there are too few samples.
When I do two MBL in a row without heatbed, the MAX(ABS(x)) of all the values is below ~0.02.
When I compare it to a MBL painfully performed (it had to complete to get the values) with the PWM regulating, I got ~0.12.

This preliminary test tends to confirm my initial assumption that even on a fully functionnal printer, the PWM can impact the first layer quality, at least on the 12V printers.

EDIT: To be precise, it is the MAX(ABS(x)) of the substracted values between two samples. This in order to compare the difference between two MBL.

**Comment by jerm** [comment]
@leptun So, it was working well _enough_ to do prints and they were coming out fine, so I moved it into my Lack Rack, at which point it (after a z-cal) immediately started failing reliably at the second (3x3) test point... something something "no signal detected". I re-re-re-ran full xyz and this time it started grabbing the paper during the no-sheet Z-calibration, so I lowered my pinda (it's now precariously close to on-level with the nozzle, but not quite), and now it's working fine again.... though my live-z is -1.405 for the smooth sheet 😳 

I'll probably order another pinda in case this one is going. Thanks for all your help!

**Comment by leptun** [comment]
@jerm At least you figured out what the problem was. Can’t you raise the pinda again now that XYZ cal. passed? Or is it crashing immediately if you do.

Also, XYZ cal. only needs tp be run after you assemble the printer as the values don’t really change when you move the pinda. Z calibration should have helped you more.

**Comment by Kachidoki2807** [comment]
I am using the patched firmware since a week now. It does the job without an issue.
I hope the code will be reviewed and a merge will be done into the official repo.
I will leave my MK2.5S to a friend in approx. 2 weeks, telling him to not update the firmware until this issue is officially solved because in state, the machine is unusable with the v3.8.1-RC1.

**Comment by jefftroll** [comment]
I only have this issue on my MK3S when attempting to probe a smooth steel sheet with buildtak pei sticker on it.  My powder coated prusa sheet that came with the printer probes every time with no fails.  I put this steel sheet plus build tak pei and it fails every other print to probe but eventually passes and prints wonderfully.  Its not a level issue that I can tell with the sheet as I have done a 59 hour full plate print with no issues.  If you have a powder coated sheet try the 7x7 on it vs the smooth PEI.
I'll probably eventually order a stock prusa smooth steel sheet and see if it is just this piece of steel it does not like.

**Comment by andreyman3d2k** [comment]
Hi - I just started experiencing this issue as well - I3 MK3 running 3.8.1 - nothing extraordinary happened with the printer, it just.. started doing it. Throwing up the error at the third calibration point. I tried a different sheet (the powdercoated vs. regular smooth), same thing. I've read through this entire thread but I can't make heads or tails of it - was a cause/solution identified? Sorry, I'm not terribly technical and pretty new to 3D printing. I read on another thread someone said it was a faulty PINDA probe? Would appreciate any help!

- Andrey

EDIT: Reached out to Prusa support and on their suggestion performed a factory reset, followed immediately (before initial calibration) by latest firmware flash, followed by calibration. This resolved the issue.

**Comment by EasyAsABC123** [comment]
Just updated printer to MK3s and to FIRMWARE 3.8.1, it now fails with infinite loop for Z calibration when the bed is heated...but passed the wizard.  What should i do?

I download FW381-RC1-Build2851.zip, I upgraded my MK3 -> MK3s which hex should i use?

**Comment by cimoalpacino** [comment]
> 
> 
> Just updated printer to MK3s and to FIRMWARE 3.8.1, it now fails with infinite loop for Z calibration when the bed is heated...but passed the wizard. What should i do?
> 
> I download FW381-RC1-Build2851.zip, I upgraded my MK3 -> MK3s which hex should i use?

@EasyAsABC123 Why have you downloaded the RC1 build?

Use the stable version ([https://github.com/prusa3d/Prusa-Firmware/releases/download/v3.8.1/prusa3d_fw_MK3S_3_8_1_2869.hex](https://github.com/prusa3d/Prusa-Firmware/releases/download/v3.8.1/prusa3d_fw_MK3S_3_8_1_2869.hex))

**Comment by ekobres** [comment]
> @Kachidoki2807 Thanks for testing. Here are the changes: https://github.com/leptun/Prusa-Firmware/tree/MK3_BED_LEVELING_PWM
> In theory it should disable the "disable" right after the function that probes returns. This was just a quick test to see if this implementation actually fixes the problem. The only places that might need checking is what happens if `kill` is triggered during probing. It might cause undesired behavior. Testing the safety aspects is definitely needed.

Great job on the fix. Any idea if Prusa will merge this into production? I'm having this issue consistently.

**Comment by leptun** [comment]
@ekobres it will probably get merged in 3.9.x

**Comment by doxie** [comment]
I just upgraded my 2s to 2.5 with an upgrade package I've had sitting around for a year. I couldn't complete a single print, it failed the z leveling on the rear middle position. Downgraded firmware and it stutters at the same position but now prints.

**Comment by ekobres** [comment]
> I just upgraded my 2s to 2.5 with an upgrade package I've had sitting around for a year. I couldn't complete a single print, it failed the z leveling on the rear middle position. Downgraded firmware and it stutters at the same position but now prints.

See if adding startup gcode to disable the bed heater during mesh leveling (G80 is the mesh level in your startup gcode) fixes your issue as a workaround. I haven’t updated firmware, but found the gcode fix to be pretty effective as a temporary measure.

**Comment by 3d-gussner** [comment]
Is this still an issue or can this be closed?
We improved the MBL meanwhile and I think this issue can be closed.
[stale][fixed]

**Comment by Nippylicious** [comment]
> 
> 
> Is this still an issue or can this be closed?
> We improved the MBL meanwhile and I think this issue can be closed.

For my part this topic can be close.
I'm on FW 3.8 (no time for 3.9 update) and have no issues for many hours of printing time :)

**Comment by 3d-gussner** [comment]
@Nippylicious Thanks for the fast feedback 🥇

**Comment by Kachidoki2807** [comment]
The fix work like a charm.

**Comment by ergo22** [comment]
I can also confirm that 3.9.0 did finally fix this issue for me. Before that for a long while I had to use a custom start code to stop bed heating while MBL as otherwise the success rate of starting a job was like  1 out of 10.

Thank you for the efforts and supporting the older printers.

**Comment by JohnnyDeer** [comment]
I´m closing this issue as solved. Thank you all.
